### ISSUE TITLE
GameEvent: impossible to get specific event variables without explicitly defining their type

### ISSUE THREAD

**Issue by KirillMysnik** [Source-Python-Dev-Team/Source.Python#161]
The following code is tested on CS:GO
```python
from core import echo_console
from events import Event


@Event('player_death')
def on_player_death(game_event):
    # Works
    echo_console("game_event['userid']: {}".format(game_event['userid']))

    # Works
    echo_console("game_event.get_int('dominated'): {}".format(
        game_event.get_int('dominated')))

    # KeyError: "Key 'dominated' does not exist."
    echo_console("game_event['dominated']: {}".format(game_event['dominated']))
```

Same goes for `revenge` variable. Both of them are `short` variables (though they can contain either 0 or 1). Maybe there's an issue with `short` event variables?

**Comment by satoon101** [comment]
I imagine the issue is because the variables don't exist in the KeyValues object that we use when calling `__getitem__` on the GameEvent object.  Try printing out `game_event.variables.as_dict()`, and I bet that dominated and revenge are not in there.

I do wonder, though, if they do exist in the KeyValues when they are 1, as in when you start dominating another player, `game_event['dominated']` works.

**Comment by invincibleqc** [comment]
`GameEvent.__getitem__` is reading the `m_pVariables` variable of the `CGameEvent` instance and this is up to the game to set those values, or not. After looking into IDA, the `player_death` event is fired into `CCSGameRules::DeathNotice` and the variables `revenge`, `assister` and `dominated` are only set if they are available. Meaning that they are not into the variables otherwise. The fact that `GameEvent.get_int` is not raising and returning 0 is caused cause the getter methods will never raise. They all have a second argument defining a default value if the specified key is not available. For example, the following:

```python
dominated = game_event.get_int('dominated', 9)
```

Would returns 9 instead of 0 (0 being the defaulted default value for [GameEvent.get_int](https://github.com/Source-Python-Dev-Team/Source.Python/blob/master/src/core/modules/events/events_wrap.cpp#L96)).

In conclusion, this is up to the game to set the variables and there is nothing we can do. The best solution I can propose is to catch the exception:

```python
try:
    dominated = bool(game_event['dominated'])
except KeyError:
    dominated = False
```

Or to use the explicit cast:

```python
dominated = game_event.get_bool('dominated')
```

**Comment by KirillMysnik** [comment]
Okay, thanks for replies guys. I'll go with explicit cast.
